### PR TITLE
Adding ExecuteOnReinstall attribute for SystemTools - TemplateFile

### PR DESCRIPTION
**Pull request by wojwal**
Hi there ;)

I needed this feature - in case of same MSI reinstallation or repair - my template file was not changed. 

Now it's working fine - I tested it... Pls merge the changes. Thanks ;)

### CODE REVIEW

File: src/CustomActions/SystemTools/TemplateFiles.cpp [hunk 1]
```diff
@@ -47,10 +47,13 @@ CA_API UINT __stdcall TemplateFiles_Immediate(MSIHANDLE hInstall)
             bool execute_per_component_install = (component_id.empty() || msiInstall.IsComponentInstalling(component_id));
             // execute on uninstall
             bool execute_per_component_uninstall = (component_id.empty() || msiInstall.IsComponentUnInstalling(component_id));
+            // execute on reinstall
+            bool execute_per_component_reinstall = (component_id.empty() || msiInstall.IsComponentReInstalling(component_id));
 
             bool execute = execute_per_condition && (
                 (execute_per_component_install && (attributes & ExecuteOnInstall) && msiInstall.IsInstalling()) 
                 || (execute_per_component_uninstall && (attributes & ExecuteOnUnInstall) && msiInstall.IsUnInstalling())
+				|| (execute_per_component_reinstall && (attributes & ExecuteOnReInstall) && msiInstall.IsReInstalling())
```
Comment (dblock) [inline src/CustomActions/SystemTools/TemplateFiles.cpp]: Something going on here with tabs/spaces.

**Comment by dblock** [comment]
It's good. It needs a CHANGELOG entry please (via `--amend`, squashed), fix the tab/space thing. See i you can write a test for this one, too, but I would merge it without.

**Comment by wojwal** [comment]
Ok - fixed the tabs and added changelog...

**Comment by dblock** [comment]
Could you squash these commits please?

**Comment by wojwal** [comment]
Done

**Comment by dblock** [comment]
Alright, holding.

**Comment by wojwal** [comment]
Nope - all is fine - you can merge...

I had another issue - the file path I am templating need to have double slashes (and I'm using [INSTALLDIR] in WiX). But it seems the author of the project I'm working on just modified msiext for double slashes and made custom build. So now when I build from master branch it's single slash again. 

I may create another pull request shortly that will add "DoubleSlash" option for TemplateFileProperty

**Comment by dblock** [comment]
DoubleSlash sounds like a hack, doubt it will fly here ;)

**Comment by dblock** [comment]
Merged, thanks for contributing! I will plan to do a release soon, haven't been using this project in a long time.

**Pull request merged**